### ISSUE TITLE
Support for sub aggregations

### ISSUE THREAD

**Issue by joanlopez** [xataio/xata-go#47]
Right now, it looks like the only way to build a [`AggExpressionTopValues`](https://github.com/xataio/xata-go/blob/main/xata/internal/fern-workspace/generated/go/agg_expression_top_values.go#L7) is with [`NewTopValuesAggExpression`](https://github.com/xataio/xata-go/blob/main/xata/search_filter_utils.go#L335), which expects a [`TopValuesAgg`](https://github.com/xataio/xata-go/blob/main/xata/search_filter_utils.go#L328). 

However, the latter doesn't accept any other aggregation (only column name and size), in contrast to the underlying [`TopValuesAgg`](https://github.com/xataio/xata-go/blob/main/xata/internal/fern-workspace/generated/go/top_values_agg.go#L9). Thus, it's not possible to perform a query like: 

```
{
  "aggs": {
    "topOpenings": {
      "topValues": {
        "column": "opening",
        "size": 5,
        "aggs": {
          "winRatio": {
            "average": {
              "column": "win",
            },
          }
        }
      }
    }
  }
}
```

Any plans? ETA? Thanks!

I would even be happy to contribute with the minimum viable set of changes to make the exposed `TopValuesAgg` type to be equals (or point to) the internal one. However, I could understand that you may want to revisit that internal/external types divergence in a wider and/or more generic way.

**Comment by philkra** [comment]
Hi @joanlopez , 
thank you for reporting this issue. You are correct, `TopValuesAgg` needs an `Aggs` property that accepts a value of type `AggExpressionMap`. 

A contribution to fix this would be amazing! I appreciate your thoughtfulness of divergence, the SDK is still alpha and we're still learning how to generate internal types and expose them best, keep everything in sync, etc. The fix takes precedence :)

If you open a PR, please mark me as a reviewer. Happy to guide you.

Cheer,
Philip

**Comment by philkra** [comment]
I put a PR to provide support for nested aggregations. I'll address the other two aggregations and will cut a release, in the coming days.

**Comment by philkra** [comment]
@joanlopez  the nested aggregations got released with [v0.0.2](https://github.com/xataio/xata-go/releases/tag/v0.0.2). The query from https://github.com/xataio/xata-go/issues/47#issue-2044227253 can be achieved via:

```go
searchClient, _ := xata.NewSearchAndFilterClient()
result, err := searchClient.Aggregate(ctx, xata.AggregateTableRequest{
    TableName: "<your-table-name>,
    Payload: xata.AggregateTableRequestPayload{
        Aggregations: xata.AggExpressionMap{
	    "topOpenings": xata.NewTopValuesAggExpression(xata.TopValuesAgg{
	        Column: "openings",
		Size:   xata.Int(5),
		Aggs: &xata.NestedAggsMap{
		    "winRatio": xata.NewAverageAggExpression("win"),
		},
	    }),
	},
    },
})
```

Let me know if you face any issues.

**Comment by joanlopez** [comment]
Awesome, that was considerably fast! I even didn't have time to find a spot within my spare time, great to see this evolving! :)
